src/any_agent/evaluation/llm_judge.py

```diff
@@ -0,0 +1,132 @@
+from typing import Any
+
+from litellm import acompletion
+from litellm.utils import supports_response_schema
+from pydantic import BaseModel
+
+from any_agent.config import AgentFramework
+from any_agent.evaluation.schemas import AgentOutput
+from any_agent.evaluation.tools import EvaluationTools
+from any_agent.tracing.agent_trace import AgentTrace
+from any_agent.utils.asyncio_sync import run_async_in_sync
+
+DEFAULT_PROMPT = """Please evaluate the following agent execution trace:
+
+AGENT TRACE:
+{trace_text}
+
+EVALUATION QUESTION:
+{question}"""
+
+
+LLM_JUDGE_SYSTEM_PROMPT = """You are an expert evaluator that analyzes agent execution traces to answer specific questions about agent performance and behavior.
+
+You will be provided with:
+1. A detailed trace of an agent's execution showing all the steps it took
+2. A specific evaluation question to answer
+
+Your task is to carefully analyze the trace and provide a judgment on whether the agent's performance meets the criteria specified in the question.
+
+TRACE FORMAT:
+The trace shows the conversation flow between the user, assistant (agent), and any tools used. Each message includes:
+- Role: system, user, assistant
+- Content: The actual message or tool execution details
+
+EVALUATION GUIDELINES:
+- Be objective and thorough in your analysis
+- Focus on what actually happened in the trace, not what should have happened
+- Consider the agent's reasoning, tool usage, and final output
+- If the question asks about specific actions, look for evidence of those actions in the trace
+- If unsure, err on the side of being more critical rather than lenient
+
+Answer with:
+1. "passed": true or false
+2. "reasoning": Brief explanation for your decision (2-3 sentences max)
+
+Your response MUST be valid JSON matching this schema:
+{
+  "passed": boolean,
+  "reasoning": "string"
+}"""
+
+
+class LlmJudge:
```

Comment:
No strong opinion:

In my mind, LlmJudge was a much less opinionated API: just passing (content, assertion) and letting the user decide on those values.

The current implementation is basically AgentJudge with no loop or tools (so just worse?).

I would prefer if LlmJudge could be focused on simple natural language assertions for arbitrary chunks of text, and we document to the users how to get some of those useful chunks (by documenting examples for final_output, and maybe even extracting part of the trace)